syntax = "proto3";

package opa.threading;
import "google/protobuf/any.proto";

enum MessageType {
  InitRequest = 0;
  DataRequest = 1;
  ResultOutput = 2;
  Status = 3;
  DataResponse = 4;
  InitResponse = 5;
}

message JobMsg {
  int32 job_id = 1;

  int64 job_nonce = 2;

  MessageType type = 3;

  int32 data_id = 4;

  string host = 5;

  // whether we run on same as host
  bool same_vm = 6;
  google.protobuf.Any any = 7;
}

enum StatusCode {
  Ok = 0;
  ChangedJob = 1;
  NoMore = 2;
  Over = 3;
}

message StatusMsg { StatusCode status = 1; }
